pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -138,7 +139,12 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *pav1alpha1.PodAutosc
 	// If SKS is not ready — ensure we're not becoming ready.
 	// TODO: see if we can perhaps propagate the SKS state to computing active status.
 	if !sks.IsReady() {
+		logger.Debug("SKS is not ready, marking dependencies not ready")
 		ready = 0
+		pa.Status.MarkDependenciesNotReady("DependenciesNotReady", "SKS is provisioning")
+	} else {
+		logger.Debug("SKS is ready, marking dependencies ready")
+		pa.Status.MarkDependenciesReady()
```

Comment:
Thinking about this at a meta level for a moment...  Are you sure you want a single condition for all dependencies? 
 In general, I'd probably advocate for a condition per dependency.

I comment on that here because one of the problems you have with an N-ary condition is: Who should mark it ready?

The logic we have for managing the actual "Ready" condition sort of solves this already, assuming you have a sub-condition per dependency.